### ISSUE TITLE
FN S1168: Support IndexerDeclaration and ConversionOperatorDeclaration

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#6494]
### Description

The rule [S1168](https://sonarsource.github.io/rspec/#/rspec/S1168/csharp) (ReturnEmptyCollectionInsteadOfNull) does not raise for IndexerDeclaration and ConversionOperatorDeclaration SyntaxKinds.

### Repro steps

```cs
class MyClass {
    public IEnumerable<string> this[int i] => null; // FN
    public static implicit operator List<int>(MyClass c) => null; // FN
}
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule is not raising an issue.

**Comment by mary-georgiou-sonarsource** [comment]
This issue will be tracked in [NET-1252](This issue will be tracked in [NET-357](https://sonarsource.atlassian.net/browse/NET-357)  (no publicly accessible)  (no publicly accessible)
